security/pkg/credentialfetcher/fetcher.go

```diff
@@ -0,0 +1,36 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// credentailfetcher fetches workload credentials through platform plugins.
+package credentialfetcher
+
+import (
+	"fmt"
+
+	"istio.io/istio/pkg/security"
+	"istio.io/istio/security/pkg/credentialfetcher/plugin"
+)
+
+func NewCredFetcher(credtype, trustdomain, jwtPath string) (security.CredFetcher, error) {
```

Comment:
Please move this to pilot_agent/main.go - where we process the env variables. It shouldn't init mock - that should be only done in tests, and as we agreed the JWT path should be used in 1.7, so you only need to check an env and 
create the plugin for GCE.

Reply:
I already have the credFetcherTypeEnv defined in pilot_agent/main.go. Mock is for testing purpose, and only initiated in the testing code.

Reply:
Sure, my point is that testign code should new plugin.CreateMockPlugin() and set it in security.Options, so the mock code is not linked in the prod build.